site/app/controllers/calendar/CalendarController.php

```diff
@@ -27,8 +32,179 @@ class CalendarController extends AbstractController {
+     * @Route("/courses/{_semester}/{_course}/calendar")
+     */
+    public function viewCourseCalendar(): WebResponse {
+        $calendar_messages = [];
+        $user = $this->core->getUser();
+        $courses = $this->core->getQueries()->getCourseForUserId($user->getId());
+        $name = $this->core->getConfig()->getCourse();
+        // Initialize $oneCourse and get the specific course for the course calendar
+        $oneCourse = $courses[0];
+        foreach ($courses as $course) {
+            if ($course->getTitle() === $name) {
+                $oneCourse = [$course];
+                break;
+            }
+        }
+        $gradeables = GradeableUtils::getGradeablesFromUserAndCourse($this->core, $calendar_messages);
+
+        return new WebResponse(CalendarView::class, 'showCalendar', CalendarInfo::loadGradeableCalendarInfo($this->core, $gradeables, $oneCourse, $calendar_messages), true);
+    }
+
+    /**
+     * @Route("/courses/{_semester}/{_course}/calendar/items/new", methods={"POST"})
+     * @AccessControl(role="INSTRUCTOR")
+     */
+    public function createMessage(): RedirectResponse {
+        // Checks if the values exist that are set and returns an error message if not
+        if (isset($_POST['type'])) {
+            $type = $_POST['type'];
+        }
+        else {
+            $this->core->addErrorMessage("Invalid or incorrect type given");
+            return new RedirectResponse($this->core->buildCourseUrl(['calendar']));
+        }
+
+        if (isset($_POST['date'])) {
+            $date = $_POST['date'];
+        }
+        else {
+            $this->core->addErrorMessage("Invalid or incorrect date given");
+            return new RedirectResponse($this->core->buildCourseUrl(['calendar']));
+        }
+
+        if (isset($_POST['text'])) {
+            $text = $_POST['text'];
+        }
+        else {
+            $this->core->addErrorMessage("Invalid or incorrect text given");
+            return new RedirectResponse($this->core->buildCourseUrl(['calendar']));
+        }
+
+        if (strip_tags($text) !== $text) {
+            $this->core->addErrorMessage("HTML cannot be used in this text");
+            return new RedirectResponse($this->core->buildCourseUrl(['calendar']));
+        }
+
+        $calendar_item = new CalendarItem();
+        try {
+            $calendar_item->setStringType($type);
+        }
+        catch (\InvalidArgumentException $e) {
+            $this->core->addErrorMessage("That is not a valid calendar item type");
+            return new RedirectResponse($this->core->buildCourseUrl(['calendar']));
+        }
+        $calendar_item->setDate(new \DateTime($date));
+        try {
+            $calendar_item->setText($text);
+        }
+        catch (\InvalidArgumentException $e) {
+            $this->core->addErrorMessage("Text exceeds 255 characters which is not allowed");
+            return new RedirectResponse($this->core->buildCourseUrl(['calendar']));
+        }
```

Comment:
Same here.  You technically don't know that the argument is invalid because it's more than 255 characters long.  All you know is that `setText()` received an invalid argument.  Yes, the length limit is currently the only limitation, but we want to remain open to future development in which more error cases could be added.

Just return the error message in the exception.